src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -309,6 +309,7 @@ def _retry_run(func: Callable, *args, **kwargs) -> Any:
 def worker_process(
     partition_queue: Union[mp.Queue, queue.Queue],
     parquet_queue: Union[mp.Queue, queue.Queue],
+    process_or_thread_error_indicator: Union[mp.Queue, queue.Queue],
```

Comment:
do we need another queue, can we reuse the existing parquet_queue, using another constant to signal that the process has exit because of running out of tasks?